internal/streamingnode/server/wal/interceptors/timetick/timetick_interceptor.go

```diff
@@ -0,0 +1,158 @@
+func (impl *timeTickAppendInterceptor) executeSyncTimeTick(interval time.Duration, param interceptors.InterceptorBuildParam) {
+	underlyingWALImpls := param.WALImpls
+
+	logger := log.With(zap.Any("channel", underlyingWALImpls.Channel()))
+	logger.Info("start to sync time tick...")
+	defer logger.Info("sync time tick stopped")
+
+	// Send first timetick message to wal before interceptor is ready.
+	for count := 0; ; count++ {
+		// Sent first timetick message to wal before ready.
+		// New TT is always greater than all tt on previous streamingnode.
+		// A fencing operation of underlying WAL is needed to make exclusive produce of topic.
+		// Otherwise, the TT principle may be violated.
+		// And sendTsMsg must be done, to help ackManager to get first LastConfirmedMessageID
+		// !!! Send a timetick message into walimpls directly is safe.
+		select {
+		case <-impl.ctx.Done():
+			return
+		default:
+		}
+		if err := impl.sendTsMsg(impl.ctx, underlyingWALImpls.Append); err != nil {
+			log.Warn("send first timestamp message failed", zap.Error(err), zap.Int("retryCount", count))
+			// TODO: exponential backoff.
+			time.Sleep(50 * time.Millisecond)
+			continue
+		}
+		break
+	}
+
+	// interceptor is ready now.
+	close(impl.ready)
+	logger.Info("start to sync time ready")
+
+	// interceptor is ready, wait for the final wal object is ready to use.
+	wal := param.WAL.Get()
+
+	// TODO: sync time tick message to wal periodically.
+	// Add a trigger on `AckManager` to sync time tick message without periodically.
+	// `AckManager` gather detail information, time tick sync can check it and make the message between tt more smaller.
+	ticker := time.NewTicker(interval)
+	defer ticker.Stop()
+	for {
+		select {
+		case <-impl.ctx.Done():
+			return
+		case <-ticker.C:
+			if err := impl.sendTsMsg(impl.ctx, wal.Append); err != nil {
+				log.Warn("send time tick sync message failed", zap.Error(err))
+			}
+		}
+	}
+}
+
+// syncAcknowledgedDetails syncs the timestamp acknowledged details.
+func (impl *timeTickAppendInterceptor) syncAcknowledgedDetails() {
+	// Sync up and get last confirmed timestamp.
+	ackDetails, err := impl.ackManager.SyncAndGetAcknowledged(impl.ctx)
+	if err != nil {
+		log.Warn("sync timestamp ack manager failed", zap.Error(err))
+	}
+
+	// Add ack details to ackDetails.
+	impl.ackDetails.AddDetails(ackDetails)
+}
+
+// sendTsMsg sends first timestamp message to wal.
+// TODO: TT lag warning.
+func (impl *timeTickAppendInterceptor) sendTsMsg(_ context.Context, appender func(ctx context.Context, msg message.MutableMessage) (message.MessageID, error)) error {
+	// Sync the timestamp acknowledged details.
+	impl.syncAcknowledgedDetails()
+
+	if impl.ackDetails.Empty() {
+		// No acknowledged info can be sent.
+		// Some message sent operation is blocked, new TT cannot be pushed forward.
+		return nil
+	}
+
+	// Construct time tick message.
+	msg, err := newTimeTickMsg(impl.ackDetails.LastAllAcknowledgedTimestamp(), impl.sourceID)
```

Comment:
Here, we could ignore the error acknowledgement, may be skip some timtick sent, but I think it's make no sense. 
Just drop the illegal message at scanner side based on timetick rule is best error-handling.